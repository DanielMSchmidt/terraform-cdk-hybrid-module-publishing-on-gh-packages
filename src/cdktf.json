{
  "language": "typescript",
  "app": "npx ts-node tfModules.ts",
  "terraformProviders": [],
  "terraformModules": [
    "terraform-aws-modules/s3-bucket/aws",
    "terraform-aws-modules/s3-bucket/aws//modules/notification"
  ],
  "output": "modules",
  "projectId": "425d04d6-b6d2-4aaa-9c58-438eac4ea6fc"
}
