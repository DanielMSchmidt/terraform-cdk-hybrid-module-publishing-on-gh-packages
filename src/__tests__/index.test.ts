import { AwsProvider } from "@cdktf/provider-aws";
import { LambdaFunction } from "@cdktf/provider-aws/lib/lambdafunction";
import { Testing } from "cdktf";
import "cdktf/lib/testing/adapters/jest";
import { S3Bucket } from "..";

// To learn more about testing see cdk.tf/testing
describe("MyConstruct", () => {
  it("should synthesize", () => {
    expect(
      Testing.synthScope((scope) => {
        new AwsProvider(scope, "aws", {
          region: "us-east-1",
        });
        new S3Bucket(scope, "my-bucket", {
          bucket: "my-bucket",
        });
      })
    ).toMatchSnapshot();
  });

  it("can add a lambda trigger", () => {
    expect(
      Testing.synthScope((scope) => {
        new AwsProvider(scope, "aws", {
          region: "us-east-1",
        });
        const bucket = new S3Bucket(scope, "my-bucket", {
          bucket: "my-bucket",
        });
        const lambda = new LambdaFunction(scope, "my-lambda", {
          functionName: "my-lambda",
          role: "my-role",
        });

        bucket.notifyLambda(lambda, "Post", {
          filterPrefix: "my-upload",
          filterSuffix: ".png",
        });
      })
    ).toMatchSnapshot();
  });
});
