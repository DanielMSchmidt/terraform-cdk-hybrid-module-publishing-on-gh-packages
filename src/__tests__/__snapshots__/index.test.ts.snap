// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyConstruct can add a lambda trigger 1`] = `
"{
  \\"module\\": {
    \\"my-bucket_019C00C9\\": {
      \\"bucket\\": \\"my-bucket\\",
      \\"force_destroy\\": true,
      \\"source\\": \\"terraform-aws-modules/s3-bucket/aws\\"
    },
    \\"my-bucket_notification-0_F3E56F61\\": {
      \\"bucket\\": \\"\${module.my-bucket_019C00C9.s3_bucket_id}\\",
      \\"eventbridge\\": true,
      \\"lambda_notifications\\": [
        {
          \\"lambda\\": {
            \\"event_type\\": \\"s3:ObjectCreated:Post\\",
            \\"filterPrefix\\": \\"my-upload\\",
            \\"filterSuffix\\": \\".png\\",
            \\"function_arn\\": \\"\${aws_lambda_function.my-lambda.arn}\\",
            \\"function_name\\": \\"\${aws_lambda_function.my-lambda.function_name}\\"
          }
        }
      ],
      \\"source\\": \\"terraform-aws-modules/s3-bucket/aws//modules/notification\\"
    }
  },
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_lambda_function\\": {
      \\"my-lambda\\": {
        \\"function_name\\": \\"my-lambda\\",
        \\"role\\": \\"my-role\\"
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"4.12.1\\"
      }
    }
  }
}"
`;

exports[`MyConstruct should synthesize 1`] = `
"{
  \\"module\\": {
    \\"my-bucket_019C00C9\\": {
      \\"bucket\\": \\"my-bucket\\",
      \\"force_destroy\\": true,
      \\"source\\": \\"terraform-aws-modules/s3-bucket/aws\\"
    }
  },
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"4.12.1\\"
      }
    }
  }
}"
`;
